### PR TITLE
fix(chat): add error messages for failed publish requests (Issue #1481)

### PR DESCRIPTION
**Pull request by Gimir**
**Description:**

Add receiving error messages through Next proxy api and displaying them in failed publish requests: approve, create and rulesList

Issues:

- Issue #1481

**UI changes**

Снимок экрана 2024-07-29 в 18 39 32

Снимок экрана 2024-07-29 в 18 40 00

**Checklist:**

- [x] the pull request name complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] the pull request name starts with `fix(<scope>):`, `feat(<scope>):`, `feature(<scope>):`, `chore(<scope>):`, `hotfix(<scope>):` or `e2e(<scope>):`. If contains breaking changes then the pull request name must start with `fix(<scope>)!:`, `feat(<scope>)!:`, `feature(<scope>)!:`, `chore(<scope>)!:`, `hotfix(<scope>)!:` or `e2e(<scope>)!:` where `<scope>` is name of affected project: `chat`, `chat-e2e`, `overlay`, `shared`, `sandbox-overlay`, etc.
- [x] the pull request name ends with `(Issue #<TICKET_ID>)` (comma-separated list of issues)
- [x] I confirm that do not share any confidential information like API keys or any other secrets and private URLs

### CODE REVIEW

**Comment by Gimir** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1840.nightly-test.deltixhub.io
>E2E tests status: failed

File: apps/chat/src/utils/server/get-response-error-msg.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { Response } from 'node-fetch';
+
+export async function getResponseErrorMsg(res: Response) {
```
Comment (IlyaBondar) [inline apps/chat/src/utils/server/get-response-error-msg.ts]: Let's put this method into `ServerUtils`

File: apps/chat/src/utils/server/api.ts [hunk 2]
```diff
@@ -107,10 +107,14 @@ export class ApiUtils {
     }).pipe(
       switchMap((response) => {
         if (!response.ok) {
-          return throwError(() => new Error(response.status + ''));
+          return from(response.text()).pipe(
```
Comment (IlyaBondar) [inline apps/chat/src/utils/server/api.ts:110]: I'm not sure that we always will receive some text.
At least it would be nice to add `catch` also

File: apps/chat/src/utils/server/api.ts [hunk 3]
```diff
@@ -107,10 +107,14 @@ export class ApiUtils {
...
+            switchMap((errorMessage) =>
+              throwError(() => new Error(errorMessage)),
+            ),
+          );
         }
 
-        return from(response.json());
+        return response.json();
```
Comment (IlyaBondar) [inline apps/chat/src/utils/server/api.ts]: Why did you remove `from` here?

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1840.nightly-test.deltixhub.io
>E2E tests status: failed

File: apps/chat/src/utils/server/api.ts (hunk 3 above)
Comment (Gimir) [reply apps/chat/src/utils/server/api.ts]: Was trying another way of handling error and forgot to revert the changes

**Comment by Gimir** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1840.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by Gimir** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1840.nightly-test.deltixhub.io
>E2E tests status: failed

File: apps/chat/src/utils/server/server.ts [hunk 4]
```diff
@@ -16,4 +18,23 @@ export class ServerUtils {
...
+    } catch (err) {
+      resBody = null;
+    }
+    try {
+      msg = (await res.json()) as string;
+    } catch (err) {
+      msg = resBody;
+    }
+
+    return msg;
```
Comment (IlyaBondar) [inline apps/chat/src/utils/server/server.ts]:
```suggestion  
   try {
      return (await res.json()) as string;
    } catch {
      try {
        return await res?.text();
      } catch {
        return null;
      }
    }
```

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1840.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by Gimir** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1840.nightly-test.deltixhub.io
>E2E tests status: success

**Review by IlyaBondar — APPROVED** [review]
